AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DiscardChangesDialog.kt

```diff
@@ -17,21 +17,21 @@ package com.ichi2.anki.dialogs
 
 import android.content.Context
 import androidx.appcompat.app.AlertDialog
-import com.ichi2.anki.R
 import com.ichi2.utils.message
 import com.ichi2.utils.negativeButton
 import com.ichi2.utils.positiveButton
 import com.ichi2.utils.show
 
-class DiscardChangesDialog {
-    companion object {
-        fun showDialog(
-            context: Context,
-            positiveMethod: () -> Unit
-        ) = AlertDialog.Builder(context).show {
-            message(R.string.discard_unsaved_changes)
-            positiveButton(R.string.dialog_ok) { positiveMethod() }
-            negativeButton(R.string.dialog_cancel)
-        }
+object DiscardChangesDialog {
+    fun showDialog(
+        context: Context,
+        positiveButtonText: String = "Discard",
+        negativeButtonText: String = "Keep Editing",
+        message: String,
```

Comment:
and then remove the unnecessary parameters from the callers

```suggestion
        positiveButtonText: String = context.getString(R.string.discard),
        negativeButtonText: String = CollectionManager.TR.addingKeepEditing(),
        message: String = CollectionManager.TR.addingDiscardCurrentInput(),
```